cdl/export_route_service.go

```diff
@@ -0,0 +1,190 @@
+	Description     string               `json:"description,omitempty"`
+	DisplayName     string               `json:"displayName" validate:"required"`
+	Source          Source               `json:"source" validate:"required"`
+	AutoExport      bool                 `json:"autoExport,omitempty"`
+	Destination     Destination          `json:"destination" validate:"required"`
+	ServiceAccount  ExportServiceAccount `json:"serviceAccount" validate:"required"`
+	CreatedBy       string               `json:"createdBy,omitempty"`
+	CreatedOn       string               `json:"createdOn,omitempty"`
+	UpdatedBy       string               `json:"updatedBy,omitempty"`
+	UpdatedOn       string               `json:"updatedOn,omitempty"`
+}
+
+type Source struct {
+	CdlResearchStudy ExportResearchStudySource `json:"cdlResearchStudy" validate:"required"`
+}
+
+type Destination struct {
+	CdlResearchStudy ExportResearchStudyDestination `json:"cdlResearchStudy" validate:"required"`
+}
+
+type ExportResearchStudySource struct {
+	Endpoint string              `json:"endpoint" validate:"required"`
+	Allowed  *ExportAllowedField `json:"allowed,omitempty"`
+}
+
+type ExportResearchStudyDestination struct {
+	Endpoint string `json:"endpoint" validate:"required"`
+}
+
+type ExportAllowedField struct {
+	DataObject []ExportDataObject `json:"dataObject,omitempty"`
+}
+
+type ExportDataObject struct {
+	Type        string        `json:"type"`
+	ExportLabel []ExportLabel `json:"label,omitempty"`
+}
+
+type ExportLabel struct {
+	Name             string `json:"name,omitempty"`
+	ApprovalRequired bool   `json:"approvalRequired,omitempty"`
+}
+
+type ExportServiceAccount struct {
+	CdlServiceAccount ExportServiceAccountDetails `json:"cdlServiceAccount" validate:"required"`
+}
+
+type ExportServiceAccountDetails struct {
+	ServiceId           string `json:"serviceId" validate:"required"`
+	PrivateKey          string `json:"privateKey" validate:"required"`
+	AccessTokenEndPoint string `json:"accessTokenEndPoint" validate:"required"`
+	TokenEndPoint       string `json:"tokenEndPoint" validate:"required"`
+}
+
+type ExportRouteService struct {
+	client   *Client
+	config   *Config
+	validate *validator.Validate
+}
+
+func (exp *ExportRouteService) path(components ...string) string {
+	return path.Join(components...)
+}
+
+type ExportRouteBundleEntry struct {
+	FullURL  string      `json:"fullUrl,omitempty"`
+	Resource ExportRoute `json:"resource,omitempty"`
+}
+
+type LinkElementType struct {
+	Relation string `json:"relation"`
+	Url      string `json:"url"`
+}
+
+type ExportRouteBundleResponse struct {
+	ResourceType string                   `json:"resourceType,omitempty"`
+	Id           string                   `json:"id,omitempty"`
+	Type         string                   `json:"type,omitempty"`
+	Link         []LinkElementType        `json:"link,omitempty"`
+	Entry        []ExportRouteBundleEntry `json:"entry,required"`
```

Comment:
move `required` keyword to  `validate` label

Reply:
Thanks! I have corrected this to include it within validate tag